File: src/main/java/org/moab/command/AccountCreateCommand.java
package org.moab.command;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;

public class AccountCreateCommand implements MOABCommand {
    @Getter private int version = 0;
    @Getter private String name = "createEvent";
    @Getter @Setter private String clientName;
    @Getter @Setter private String clientID;
    @Getter @Setter private LocalDate clientDoB;

    public AccountCreateCommand(String clientName, String clientID, LocalDate dob) {
        this.clientName = clientName;
        this.clientID = clientID;
        this.clientDoB = dob;
    }
}

File: src/main/java/org/moab/command/AccountCreateCommand.kt
package org.moab.command

import java.time.LocalDate

/**
 * Created by neo on 20/4/16.
 *
 */
data class AccountCreateCommand(var clientName: String, var clientID: String, var clientDoB: LocalDate ) : MOABCommand {
    val version: Int
        get() = 0

    val name: String
        get() = "createEvent"
}
